cli/src/cli_util.rs

```diff
@@ -1312,6 +1325,51 @@ Then run `jj squash` to move the resolution into the conflicted commit."#,
 
         Ok(())
     }
+
+    /// Identifies branches which are eligible to be moved automatically during
+    /// `jj commit` and `jj new`. Whether a branch is eligible is determined by
+    /// its target and the user and repo config for "advance-branches".
+    ///
+    /// Returns a Vec of branches in `repo` that point to any of the `from`
+    /// commits and that are eligible to advance. The `from` commits are
+    /// typically the parents of the target commit of `jj commit` or `jj new`.
+    ///
+    /// Branches are not moved until
+    /// `WorkspaceCommandTransaction::advance_branches()` is called with the
+    /// `AdvanceableBranch`s returned by this function.
+    ///
+    /// Returns an empty `std::Vec` if no branches are eligible to advance.
+    pub fn get_advanceable_branches<'a>(
+        &self,
+        repo: &impl Repo,
+        from: impl IntoIterator<Item = &'a CommitId>,
+    ) -> Vec<AdvanceableBranch> {
+        let advance_branches = self.settings.advance_branches();
+        let overrides = self.settings.advance_branches_overrides();
+        let allow_branch = |branch: &str| {
+            let overridden = overrides.iter().any(|x| x.eq(branch));
+            (advance_branches && !overridden) || (!advance_branches && overridden)
+        };
```

Comment:
Maybe better to add positive/negative patterns instead?
```toml
advance-branches-includes = ["glob:*"]
advance-branches-excludes = ["exact:main"]
```

Suppose the user configured `experimental-advance-branches.enabled = true` globally, he might add `"main"` to the repo-level `overrides` list. If he changes mind later and turns it off globally, the "main" branch will start advancing. That's scary.

Reply:
Thanks for this suggestion, I agree. I updated the PR so that an example config looks like this:

```
[experimental-advance-branches]
enabled-branches = ["glob:*"]
disabled-branches = ["main"]
```